File: sqrt.c
// author: mrozinsky (Michal Jakub Mrozinski)
// 18.09.2021 

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <stdint.h> // needed for uint32_t
#include <math.h>

int main()
{

	clock_t start = clock();
	
	float result = 0;

	for (uint32_t i=1; i<10000001; i++)
	{
		result = sqrt(i);
	}
		
	clock_t end = clock();
	double execution_time = (double)(end - start) / CLOCKS_PER_SEC;
	
	printf("******************\n");
	printf("SQRT - C Version\n");
	printf("Program that calculates square roots of numbers from 1 to 10000000 (10 millions)\n");
	printf("Execution time: \n%f\n", execution_time);
	printf("******************\n");

	return 0;
	
}

File: sqrt.py
# author: mrozinsky (Michal Jakub Mrozinski)
# 18.09.2021 

# Results:
# Duron 750Mhz - 24.5478720665 seconds 
# Intel Core i7-3770 3.40 GHz (using only one thread)- 1.920609712600708 seconds
# Intel Core 2 Duo E4500 2.20 GHz (using only one thread) - 3.418095350265503 seconds

import time
import math

start = time.time()
for i in range(1, 10000001):
	tmp = math.sqrt(i)

end = time.time()

print("******************");
print("SQRT - Python Version");
print("Program that calculates square roots of numbers from 1 to 10000000 (10 millions)");
print("Execution time: \n" + str(end - start)) 
print("******************");
